cabra/src/pages/components/RelationListEdit.tsx

```diff
@@ -0,0 +1,153 @@
+import { DirectedRelationType, ITaskRelation } from "../../types/taskRelation";
+import {
+  createRelation,
+  deleteRelation,
+  getRelatedTasks,
+} from "../../api/taskRelations";
+import { deleteTask, getTask, getTasks } from "../../api/tasks";
+import tw, { styled } from "twin.macro";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+
+import { ITask } from "../../types/task";
+import MiniTaskDelete from "./MiniTaskDelete";
+import { PlusIcon } from "@heroicons/react/solid";
+
+const Container = styled.div(tw`text-gray-50 w-full space-y-2`);
+const CenteredLabel = styled.div(tw`text-gray-50 font-bold text-center`);
+const AddSubtaskButton = styled.button(
+  tw`bg-secondary p-2 text-white font-semibold`,
+  tw`rounded shadow-2xl flex gap-2 transition-transform transform hover:scale-105`
+);
+
+interface Props {
+  parentId: string;
+  relationType: DirectedRelationType;
+  relatedTasks: ITask[];
+  parentCallback: () => void;
+}
+
+export default function RelationListEdit({
+  parentId,
+  relationType,
+  relatedTasks,
+  parentCallback,
+}: Props) {
+  const client = useQueryClient();
+
+  const addRelation = useMutation(createRelation, {
+    onSuccess: () => {
+      client.invalidateQueries(
+        getRelatedTasks.cacheKey(parentId, relationType)
+      );
+    },
+  });
+
+  const removeTask = useMutation(
+    (relatedTaskId: string) => deleteTask(relatedTaskId),
+    {
+      onSuccess: () => {
+        client.invalidateQueries(getTasks.cacheKey);
+        client.invalidateQueries(getTask.cacheKey(parentId));
+        client.invalidateQueries(
+          getRelatedTasks.cacheKey(parentId, relationType)
+        );
+      },
+    }
+  );
+
+  const removeRelation = useMutation(
+    (relationId: string) => deleteRelation(relationId),
+    {
+      onSuccess: () => {
+        client.invalidateQueries(getTasks.cacheKey);
```

Comment:
Ideally, all this should go to a separate file as a custom hook. But we can leave it as it is, for now